### PR TITLE
4 implement a circuit stage

### PR DESCRIPTION
**Pull request by marcosfrenkel**
Not ready for merging yet, but in a stage ready to be shown. Variable names are not final, still finding bugs and extremely finicky about how it likes its arguments being passed.

basic example:

      reg4 ← reg 4
      reg4[1;2] ← ⊂q1
      reg4[2;2] ← ⊂(H+.×q0)
      gates ← (H 2) (CNOT (0 3)) 
      (1) gates stage (thread_reg reg4)

You are measuring qubit 1 and applying a CNOT gate with qubit 0 as control and 3 as target and an H gate to qubit 2

### CODE REVIEW

File: src/qc_ncsa.aplf [hunk 1]
```diff
@@ -211,26 +211,33 @@
         }
 
         ⍝ Measures a specific qubit in a threaded register.
-        ⍝ Returns the measured state of the qubit and the new vector state of the remaining normalized vector state of the other qubits.
+        ⍝ Returns the measured state of the qubit and the new normalized vector state vector state of all the qubits.
```
Comment (nunezco2) [inline src/qc_ncsa.aplf:214]: Remove repetition

**Review by nunezco2 — CHANGES_REQUESTED** [review]

File: src/qc_ncsa.aplf (hunk 1 above)
Comment (marcosfrenkel) [reply src/qc_ncsa.aplf:214]: This will get fixed with PR #27

**Comment by marcosfrenkel** [comment]
I found a bug when trying to apply a CNOT gate from example from qubit 4 to qubit 0, for some reason nothing seems to be happening and I could not figure out why this was the case. Maybe we could discuss it in person

**Review by nunezco2 — APPROVED** [review]
Marcos, this is great! The `stage` function is way, way simpler.

**Pull request merged**